.github/workflows/qemu.yaml

```
@@ -0,0 +1,72 @@
+name: QEMU environment
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  release:
+    types:
+      - published
+
+jobs:
+  build-linux-arm64:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Install VM dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y qemu-system-arm qemu-efi-aarch64 qemu-utils cloud-image-utils wget
+      - name: Provision VM
+        run: |
+          wget https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-arm64.img
+
+          qemu-img create -f qcow2 -F qcow2 \
+            -b jammy-server-cloudimg-arm64.img \
+            jammy-snapshot.qcow2 200G
+
+          cat >"user-data" << EOF
+          #cloud-config
+          password: passw0rd
+          chpasswd: { expire: False }
+          ssh_pwauth: True
+          EOF
+          cloud-localds user-data.img user-data
+
+          cp /usr/share/AAVMF/AAVMF_CODE.fd .
+          cp /usr/share/AAVMF/AAVMF_VARS.fd .
+
+          randmac=$(openssl rand -hex 6 | sed 's/\(..\)/\1:/g; s/:$//')
+
+          qemu-system-aarch64 \
+            -m 2048 \
+            -cpu max -smp 4 \
+            -machine virt,gic-version=max \
+            -display none \
+            -drive if=pflash,format=raw,readonly=on,file=AAVMF_CODE.fd \
+            -drive if=pflash,format=raw,file=AAVMF_VARS.fd \
+            -drive if=none,file=jammy-server-cloudimg-arm64.img,id=hd0 \
+            -drive file=user-data.img,format=raw \
+            -device virtio-blk-device,drive=hd0 \
+            -netdev type=user,id=net0,hostfwd=tcp::22225-:22 \
+            -device virtio-net-device,netdev=net0,mac=$randmac \
```

```suggestion
            -device virtio-net-device,netdev=net0 \
```

qemu will autogenerate a mac